src/index/DeltaTriples.cpp

```diff
@@ -251,26 +244,47 @@ DeltaTriples::DeltaTriples(const Index& index)
 // ____________________________________________________________________________
 DeltaTriplesManager::DeltaTriplesManager(const IndexImpl& index)
     : deltaTriples_{index},
-      currentLocatedTriplesSnapshot_{deltaTriples_.wlock()->getSnapshot()} {}
+      currentLocatedTriplesSnapshot_{deltaTriples_.wlock()->getSnapshot()} {
+  // Set up callback for update-no-snapshots parameter changes
+  // Initialize with current value to avoid deadlock in callback
+  static bool previousUpdateNoSnapshotsValue =
+      getRuntimeParameter<&RuntimeParameters::updateNoSnapshots_>();
+  globalRuntimeParameters.wlock()->updateNoSnapshots_.setOnUpdateAction(
+      [this](bool newValue) {
+        // When changed from true to false, update metadata and create a
+        // snapshot
+        if (previousUpdateNoSnapshotsValue && !newValue) {
```

Comment:
I don't quite understand this. Does this hook also fire when the same value is overwritten with the same value? If yes, please add a comment explaining under whose circumstances this would happen and why using static is safe in this case.